plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -250,16 +250,16 @@ class session final : public session_base {
                      block_to_send->blocks_result_base.this_block  = {self.current_blocks_request.start_block_num, *this_block_id};
```

Comment:
Should we break in case this_block_id has no value?

Reply:
I don't understand what you are asking. This whole block is in the if for if `this_block_id` has value.

Reply:
I could not place the comment right below the `if`. I was asking do we need an `else` to break out the outer `while` in case `this_block_id` has no value.

Reply:
Not really sure what we should do in this case. If running without ship logs and without a block log it is possible that we can't find the requested block. In that case we, maybe we should log an error and disconnect? @spoonincode